OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -108,7 +108,12 @@ public virtual void DoProduction(Actor self, ActorInfo producee, ExitInfo exitin
 
 		protected virtual Exit SelectExit(Actor self, ActorInfo producee, string productionType, Func<Exit, bool> p)
 		{
-			return self.RandomExitOrDefault(self.World, productionType, p);
+			if (self.Info.TraitInfos<ExitInfo>().FirstEnabledTraitOrDefault().RandomExit)
```

Comment:
FirstEnabledTraitOrDefault() returns a null in d2k. I dunno what other method I could use to get ExitInfo

Reply:
Returning null is intentional when there is no exit. You probably want to return null in that case (at least I think that is what the old code did).